src/cloudai/configurator/trajectory.py

```
@@ -0,0 +1,424 @@
+        for component_type in self._component_types:
+            component = components_by_type[component_type]
+            record.update(
+                {_field.name: _thaw(getattr(component, _field.name)) for _field in self._fields_by_type[component_type]}
+            )
+        return record
+
+
+def _validate_components(components: Sequence[object]) -> None:
+    non_dataclasses = [type(component).__name__ for component in components if not dataclasses.is_dataclass(component)]
+    if non_dataclasses:
+        raise TypeError(f"trajectory components must be dataclass instances: {', '.join(non_dataclasses)}")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)